content/en/tracing/trace_pipeline/ingestion_mechanisms.md

```diff
@@ -255,16 +255,292 @@ In Agent version 7.33 and forward, you can disable the rare sampler in the Agent
+    }
+}
+```
+
+Manually drop a trace:
+
+```java
+import datadog.trace.api.DDTags;
+import io.opentracing.Span;
+import datadog.trace.api.Trace;
+import io.opentracing.util.GlobalTracer;
+
+public class MyClass {
+    @Trace
+    public static void myMethod() {
+        // grab the active span out of the traced method
+        Span span = GlobalTracer.get().activeSpan();
+        // Always Drop the trace
+        span.setTag(DDTags.MANUAL_DROP, true);
+        // method impl follows
+    }
+}
+```
+
+{{< /programming-lang >}}
+{{< programming-lang lang="python" >}}
+
+Manually keep a trace:
+
+```python
+from ddtrace import tracer
+from ddtrace.constants import MANUAL_DROP_KEY, MANUAL_KEEP_KEY
+
+@tracer.wrap()
+def handler():
+    span = tracer.current_span()
+    # Always Keep the Trace
+    span.set_tag(MANUAL_KEEP_KEY)
+    # method impl follows
+```
+
+Manually drop a trace:
+
+```python
+from ddtrace import tracer
+from ddtrace.constants import MANUAL_DROP_KEY, MANUAL_KEEP_KEY
+
+@tracer.wrap()
+def handler():
+    span = tracer.current_span()
+    # Always Drop the Trace
+    span.set_tag(MANUAL_DROP_KEY)
+    # method impl follows
+```
+
+{{< /programming-lang >}}
+{{< programming-lang lang="ruby" >}}
+
+Manually keep a trace:
+
+```ruby
+Datadog::Tracing.trace(name, options) do |span, trace|
+  trace.keep! # Affects the active trace
+  # Method implementation follows
+end
+```
+
+Manually drop a trace:
+
+```ruby
+Datadog::Tracing.trace(name, options) do |span, trace|
+  trace.reject! # Affects the active trace
+  # Method implementation follows
+end
+```
+
+{{< /programming-lang >}}
+{{< programming-lang lang="go" >}}
+
+Manually keep a trace:
+
+```Go
+package main
+
+import (
+    "log"
+    "net/http"
+    "gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+    "gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+)
+
+func handler(w http.ResponseWriter, r *http.Request) {
+    // Create a span for a web request at the /posts URL.
+    span := tracer.StartSpan("web.request", tracer.ResourceName("/posts"))
+    defer span.Finish()
+
+    // Always keep this trace:
+    span.SetTag(ext.ManualKeep, true)
+    //method impl follows
+
+}
+```
+
+Manually drop a trace:
+
+```Go
+package main
+
+import (
+    "log"
+    "net/http"
+
+    "gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+    "gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+)
+
+func handler(w http.ResponseWriter, r *http.Request) {
+    // Create a span for a web request at the /posts URL.
+    span := tracer.StartSpan("web.request", tracer.ResourceName("/posts"))
+    defer span.Finish()
+
+    // Always drop this trace:
+    span.SetTag(ext.ManualDrop, true)
+    //method impl follows
+}
+```
+
+{{< /programming-lang >}}
+{{< programming-lang lang="nodejs" >}}
+
+Manually keep a trace:
+
+```js
+const tracer = require('dd-trace')
+const tags = require('dd-trace/ext/tags')
+
+const span = tracer.startSpan('web.request')
+
+// Always keep the trace
+span.setTag(tags.MANUAL_KEEP)
```

Comment:
I looked through our source code and the definition of setTag and I believe it should be used like this:

```js
span.setTag(tags.MANUAL_KEEP, true)
```

Reply:
On second look, your code should work just fine. Internally it'll set the tag value to `undefined`, which is fine as the code seems to specifically only looks to see that the value isn't exactly `false`.